herbert_core/utilities/classes/@object_lookup/object_repmat.m

```diff
@@ -0,0 +1,177 @@
+%
+%
+% Output:
+% -------
+%   obj_out     Output obj_lookup object
+
+
+% Check validity of object_array
+% ------------------------------
+if ~isscalar(obj)
+    error('HERBERT:object_lookup:invalid_argument', ...
+        'Only operates on a single object_lookup (i.e. object must be scalar');
+end
+if ~obj.filled
+    error('HERBERT:object_lookup:invalid_argument', ...
+        'The object_lookup is not initialised')
+end
+
+% Parse input
+% -----------
+% Determine number and type OK
+narg = numel(varargin);
+if ~(narg==2 || narg==3) || ~ischar(varargin{end-1})
+    error('HERBERT:object_lookup:invalid_argument', ...
+        'Invalid number or type of input argument(s)')
+end
+
+% Get iarray
+if narg==3
+    if ~is_integer_id (varargin{1}) || max(varargin{1})>obj.narray
+        error('HERBERT:object_lookup:invalid_argument', ...
+            'Index array must be contain unique integers in the range 1-%d', obj.narray)
+    end
+    iarray = varargin{1}(:);    % column vector
+else
+    iarray = (1:obj.narray)';   % column vector
+end
+
+% Get sz_repmat
+keyword = varargin{end-1};
+if numel(keyword)<2 || ~strncmpi(keyword, '-repeat', numel(keyword))
+    error('HERBERT:object_lookup:invalid_argument', ...
+        'Unrecognised keyword option %s', keyword)
+end
+sz_repmat = parse_sz_repmat (varargin{end}, numel(iarray), obj.narray); % col vector
+
+
+% Expand arrays
+% -------------
+% Get indexing arrays and sizes of stored arrays
+sz = obj.sz_;
+indx = obj.indx_;
+
+% Reshape the index arrays to the sizes of the object arrays to which repmat
+% will be applied
+sz_sel = sz(iarray);
+indx_sel = cellfun (@(x,y)reshape(x,y), indx(iarray), sz_sel, 'uniformoutput', false);
+ 
+% Now repmat the index and sz arrays as required, turning the indexd arrays back
+% into column vectors as the object definition requires
+if numel(iarray) > 1
+    if numel(sz_repmat)>1
+        sz(iarray) = cellfun(@(x,y)size_repmat(x,y), sz_sel, sz_repmat,...
+            'uniformOutput', false);
+        indx(iarray) = cellfun(@(x,y)reshape((repmat(x,y)),[],1), indx_sel, sz_repmat,...
+            'uniformOutput', false);
+    else
+        sz(iarray) = cellfun(@(x)size_repmat(x,sz_repmat{1}), sz_sel,...
+            'uniformOutput', false);
+        indx(iarray) = cellfun(@(x)reshape(repmat(x,sz_repmat{1}),[],1), indx_sel,...
+            'uniformOutput', false);
+    end
+else
+    if numel(sz_repmat)>1
+        % This case increases the number of stored object arrays from 1 to
+        % numel(sz_repmat). Therefore simply reassign sz and indx
+        sz = cellfun(@(x)size_repmat(sz{1},x), sz_repmat,...
+            'uniformOutput', false);
+        indx = cellfun(@(x)reshape(repmat(indx_sel{1},x),[],1), sz_repmat,...
+            'uniformOutput', false);
+    else
+        sz{iarray} = size_repmat(sz{1},sz_repmat{1});
+        indx{iarray} = reshape(repmat(indx_sel{1},sz_repmat{1}),[],1);
+    end
+end
```

Comment:
Hmm.... it looks like I've got no choice. The Matlab documentation explicitly states (R2018a)

"A = cellfun(func,C1,...,Cn) applies func to the contents of the cells of C1,...,Cn, so that A(i) = func(C1{i},...,Cn{i}). The function func must take n input arguments and return a scalar. The cell arrays C1,...,Cn all must have the same size."

Just in case Matlab has introduced cellarray expansion I tested for:
         c = cellfun (@horzcat, a, b, 'uniformOutput', false)
With 
      a = {[2,3], [101,102,103]}          % cell array, two elements
      b = {[11,12,13]}                       % cell array one element

I get the error message:
Error using cellfun
All of the input arguments must be of the same size and shape.
Previous inputs had size 2 in dimension 2. Input #3 has size 1

I tried with R2023a (the latest version) too; it gives the same error message.